File: utilslibhttp/src/main/java/com/tson/utils/lib/http/base/BaseApiStatus.java
package com.tson.utils.lib.http.base;

import android.support.annotation.NonNull;
import android.util.Log;

/**
 * Created tangxin
 * Time 2019/2/18 7:20 PM
 * @author tangxin
 */
public abstract class BaseApiStatus<T> implements BaseStatus<T> {

    private static final String TAG = BaseApiStatus.class.getName();

    @Override
    public void request(@NonNull String params) {
        Log.d(TAG, "JDOPushApi params【" + params + "】");
    }

    @Override
    public void before() {
    }

    @Override
    public void error(@NonNull Throwable throwable) {
    }

    @Override
    public void isEmpty() {
    }
}

File: utilslibhttp/src/main/java/com/tson/utils/lib/http/base/BaseApiStatus.kt
package com.tson.utils.lib.http.base

import android.util.Log

/**
 * Created tangxin
 * Time 2019/2/18 7:20 PM
 * @author tangxin
 */
abstract class BaseApiStatus<T> : BaseStatus<T> {

    override fun request(params: String) {
        Log.d(TAG, "params【$params】")
    }

    override fun before() {}

    override fun error(throwable: Throwable) {}

    override fun isEmpty() {}

    companion object {

        private val TAG = BaseApiStatus::class.java.name
    }
}
